src/server/utilities/constructCspHeader/index.js

```diff
@@ -0,0 +1,132 @@
+import csp from 'helmet-csp';
+import getRouteProps from '#app/routes/getInitialData/utils/getRouteProps';
+import routes from '#app/routes';
+import getOriginContext from '#contexts/RequestContext/getOriginContext';
+
+export const generateScriptSrc = (isAmp, isLive) => {
+  if (isAmp) {
+    return [
+      'https://cdn.ampproject.org',
+      'https://*.chartbeat.com',
+      'https://*.go-mpulse.net',
+      "'unsafe-inline'",
+    ];
+  }
+
+  const scriptSrc = [
+    'https://news.files.bbci.co.uk',
+    'https://*.chartbeat.com',
+    'https://*.go-mpulse.net',
+    'https://mybbc-analytics.files.bbci.co.uk',
+    'https://emp.bbci.co.uk',
+    'https://static.bbci.co.uk',
+    "'self'",
+    "'unsafe-inline'",
+  ];
+
+  if (!isLive) {
+    scriptSrc.push('https://news.test.files.bbci.co.uk');
+  }
+
+  return scriptSrc;
+};
+
+export const generateImgSrc = isLive => {
+  const imgSrc = [
+    'https://ichef.bbci.co.uk',
+    'https://ping.chartbeat.net',
+    'https://a1.api.bbc.co.uk/hit.xiti',
+    'https://news.files.bbci.co.uk',
+    'https://*.akstat.io',
+    'https://r.bbci.co.uk',
+  ];
+
+  if (!isLive) {
+    const testSrc = [
+      'https://ichef.test.bbci.co.uk',
+      'https://news.test.files.bbci.co.uk',
+      'https://logws1363.ati-host.net',
+      "data: 'self'",
+    ];
+    return imgSrc.concat(testSrc);
+  }
+  // Duplicated to maintain proper order
+  imgSrc.push("data: 'self'");
```

Comment:
Is this not duplicated right after line 49? Can line 49 not be removed?

Reply:
yes but it forces the imgSrc to be a `let` instead of a `const`. (Its not like it matters since I am modifying the actual array anyway)